### PR TITLE
Adds conversion rules for Gray and Color images wrapped by OffsetArray

### PR DESCRIPTION
**Pull request by zygmuntszpak**
This contribution is a prerequisite for [adding OffsetArrays capability to gaussian_pyramid.](https://github.com/JuliaImages/Images.jl/pull/722)

### CODE REVIEW

File: test/convert_reinterpret.jl [hunk 1]
```diff
@@ -167,6 +167,57 @@ end
...
     end
+
+    # Gray images wrapped by an OffsetArray.
+    A = rand(8,8)
+    for img in ( Gray.(A),
+                 Gray.(N0f8.(A)),
+                 Gray.(N0f16.(A)) )
+    imgo = OffsetArray(img, 0, 0)
+    s = @inferred(convert(OffsetArray{Gray{Float32},2,Array{Gray{Float32}}},imgo))
+    @test eltype(s) == Gray{Float32}
```
Comment (Evizero) [inline test/convert_reinterpret.jl]: could you also add this line to the tests

```julia
@test s isa OffsetArray{Gray{Float32},2,Array{Gray{Float32}}}
```

**Comment by zygmuntszpak** [comment]
@Evizero  @arijitkar98
This pull-request addresses the conversion issue that we encountered with  `OffsetArray` in the Gaussian pyramid pull-request.

File: test/convert_reinterpret.jl [hunk 2]
```diff
@@ -167,6 +167,57 @@ end
...
+    end
+
+    for img in ( Gray.(A),
+                 Gray.(N0f8.(A)),
+                 Gray.(N0f16.(A)) )
+    imgo = OffsetArray(img, 0, 0)
```
Comment (Evizero) [inline test/convert_reinterpret.jl]: probably would be a good idea to use something like `OffsetArray(img, -2, -1)` and then later test that the following is true

```julia
@test indices(s) === indices(imgo)
```

File: test/convert_reinterpret.jl [hunk 3]
```diff
@@ -167,6 +167,57 @@ end
...
+    s = @inferred(convert(OffsetArray{Gray{N0f8},2,Array{Gray{N0f8}}},imgo))
+    @test eltype(s) == Gray{N0f8}
+    end
+
+    for img in ( Gray.(A),
+                 Gray.(N0f8.(A)),
+                 Gray.(N0f16.(A)) )
+    imgo = OffsetArray(img, 0, 0)
```
Comment (Evizero) [inline test/convert_reinterpret.jl]: I think it would help readability a little if the content of the `for` loop is indented

**Review by Evizero — CHANGES_REQUESTED** [review]
Looks great!

Just some minor remarks concerning the tests

**Comment by Evizero** [comment]
I changed the target branch to `release-0.6`, since we do want this change in 0.6. We haven't quite yet made the hard distinction but we already somewhat started to diverge in terms of 0.6 and 0.7.

see conversation here https://github.com/JuliaImages/ImageCore.jl/pull/54

**Comment by Evizero** [comment]
I see that changing the target branch caused github to think the previous two commits on `master` should also be merged into `release-0.6`, which it shouldn't.

It might be easier - and probably cleaner - to merge it into `master` after all, and then just backport the commit

**Comment by zygmuntszpak** [comment]
I've incorporated all of the requested changes. Thanks for the great suggestions.

**Review by Evizero — APPROVED** [review]

**Comment by Evizero** [comment]
Thanks!

**Pull request merged**